### PR TITLE
Mbs 11392 runner cleanup

### PR DESCRIPTION
**Pull request by dsvoronin**
remove `shared-test` module
make stubs instead of generator functions
move stuff to testFixtures

### CODE REVIEW

File: subprojects/test-runner/client/src/test/kotlin/com/avito/test/ListWithDefault.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.avito.test
+
+internal fun <T : Any> listWithDefault(size: Int, value: T): List<T> {
+    return (0 until size).map { value }
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/client/src/test/kotlin/com/avito/test/ListWithDefault.kt:4]:
```suggestion
    return List(size) { value }
```

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/service/TestRunnerWorkAction.kt [hunk 2]
```diff
@@ -38,7 +40,13 @@ public abstract class TestRunnerWorkAction : WorkAction<TestRunnerWorkAction.Par
 
         val testResults = parameters.service.get().runTests(params, legacyTestRunParams)
 
-        factory.provideFinalizer().finalize(testResults)
+        val buildFailer: BuildFailer = BuildFailer.RealFailer()
+
+        when (val result = factory.provideFinalizer().finalize(testResults)) {
+            Finalizer.Result.Ok -> {
+            }
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/service/TestRunnerWorkAction.kt:47]:
```suggestion
                // no op
            }
```

File: subprojects/test-runner/service/build.gradle.kts [hunk 3]
```diff
@@ -32,7 +32,7 @@ dependencies {
     testImplementation(project(":common:truth-extensions"))
     testImplementation(project(":common:resources"))
     testImplementation(project(":gradle:test-project"))
-    testImplementation(project(":test-runner:shared-test"))
+    testImplementation(project(":test-runner:stub"))
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/service/build.gradle.kts:35]: I didn't get why we use a standalone module for stubs.
Do we use it only in tests?
If so, can't we share it by `testFixtures`?

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/test-runner/service/build.gradle.kts (hunk 3 above)
Comment (dsvoronin) [reply subprojects/test-runner/service/build.gradle.kts:35]: we use it in production code, i'll try to move to `device-provider`

File: subprojects/test-runner/client/src/test/kotlin/com/avito/test/ListWithDefault.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/test-runner/client/src/test/kotlin/com/avito/test/ListWithDefault.kt:4]: done in next PR, removed this extension completely in favour of stdlib fun

File: subprojects/test-runner/service/build.gradle.kts (hunk 3 above)
Comment (dsvoronin) [reply subprojects/test-runner/service/build.gradle.kts:35]: done in next PR

**Pull request merged**